titus_isolate/isolate/workload_manager.py

```diff
@@ -39,6 +42,11 @@ def __init__(self, cpu, cgroup_manager, static_cpu_allocator):
         log.info("Created workload manager")
 
     def add_workload(self, workload):
+        if self.__occupies_entire_cpu(workload, self.__cpu):
+            disable_numa_balancing()
+        else:
+            enable_numa_balancing()
```

Comment:
I think it's a little messy to enable it here blindly? Although, I understand that you'd have to keep state otherwise?

Reply:
I do this so that I can fail any previous operations around numa balancing without worry.